bin/improver-orographic-enhancement

```diff
@@ -0,0 +1,107 @@
+from improver.utilities.load import load_cube
+from improver.utilities.save import save_netcdf
+
+
+def main():
+    """Calculate orographic enhancement of precipitation from model pressure,
+    temperature, relative humidity and wind input files"""
+
+    parser = ArgParser(description="Calculate orographic enhancement.")
+
+    parser.add_argument('temperature_filepath', metavar='TEMPERATURE_FILEPATH',
+                        help='Full path to input NetCDF temperature file')
+    parser.add_argument('humidity_filepath', metavar='HUMIDITY_FILEPATH',
+                        help='Full path to input NetCDF relhumidity file')
+    parser.add_argument('pressure_filepath', metavar='PRESSURE_FILEPATH',
+                        help='Full path to input NetCDF pressure file')
+    parser.add_argument('windspeed_filepath', metavar='WINDSPEED_FILEPATH',
+                        help='Full path to input NetCDF wind speed file')
+    parser.add_argument('winddir_filepath', metavar='WINDDIR_FILEPATH',
+                        help='Full path to input NetCDF wind direction file')
+
+    parser.add_argument('orography_filepath', metavar='OROGRAPHY_FILEPATH',
+                        help='Full path to input NetCDF high resolution (1 km)'
+                        ' orography ancillary')
+
+    parser.add_argument('output_high_resolution', metavar='OUTPUT_HIGH_RES',
+                        help='Full path to write orographic enhancement file '
+                        'on high resolution (1 km) grid')
+    parser.add_argument('output_standard_grid', metavar='OUTPUT_STANDARD',
+                        help='Full path to write orographic enhancement file '
+                        'on standard grid')
+
+    parser.add_argument('--boundary_height_metres', type=float, default=1000.,
+                        help='Model height level to extract variables for '
+                        'calculating orographic enhancement, as proxy for '
+                        'the boundary layer.')
+
+    args = parser.parse_args()
+
+    # read input variables
+    bh = args.boundary_height_metres
+    constraint = iris.Constraint(height=lambda h: bh - 0.1 < h < bh + 0.1)
+
+    temperature = load_cube(args.temperature_filepath, constraint)
+    humidity = load_cube(args.humidity_filepath, constraint)
+    pressure = load_cube(args.pressure_filepath, constraint)
+    wind_speed = load_cube(args.windspeed_filepath, constraint)
+    wind_dir = load_cube(args.winddir_filepath, constraint)
+
+    # resolve u and v wind components
+    uwind, vwind = ResolveWindComponents().process(wind_speed, wind_dir)
+
+    # load 1 km orography
+    orography = load_cube(args.orography_filepath)
+
+    # calculate orographic enhancement
+    orogenh_high_res, orogenh_standard = OrographicEnhancement().process(
+        temperature, humidity, pressure, uwind, vwind, orography)
+
+    # save output files
+    save_netcdf(orogenh_high_res, args.output_high_resolution)
```

Comment:
How is this going to work in the suite configuration? I don't know if it would be easier to pass in a output directory, but we need to think about how this could work. I haven't thought it through yet.

Reply:
I've tended to be advised to supply a file path rather than a directory name, but it could be done either way.